src/wire/ipv6routing.rs

```diff
@@ -0,0 +1,664 @@
+    pub fn cmpr_i(&self) -> u8 {
+        let data = self.buffer.as_ref();
+        data[field::CMPR] >> 4
+    }
+
+    /// Return the number of prefix octects elided from the last address (addresses[n]).
+    ///
+    /// # Panics
+    /// This function may panic if this header is not the RPL Source Routing Header routing type.
+    pub fn cmpr_e(&self) -> u8 {
+        let data = self.buffer.as_ref();
+        data[field::CMPR] & 0xf
+    }
+
+    /// Return the number of octects used for padding after addresses[n].
+    ///
+    /// # Panics
+    /// This function may panic if this header is not the RPL Source Routing Header routing type.
+    pub fn pad(&self) -> u8 {
+        let data = self.buffer.as_ref();
+        data[field::PAD] >> 4
+    }
+
+    /// Return the address vector in bytes
+    ///
+    /// # Panics
+    /// This function may panic if this header is not the RPL Source Routing Header routing type.
+    pub fn addresses(&self) -> &[u8] {
+        let data = self.buffer.as_ref();
+        &data[field::ADDRESSES(data[field::LENGTH])]
+    }
+}
+
+impl<'a, T: AsRef<[u8]> + ?Sized> Header<&'a T> {
+    /// Return type specific data
+    #[inline]
+    pub fn data(&self) -> &'a[u8] {
+        let data = self.buffer.as_ref();
+        &data[field::DATA(data[field::LENGTH])]
+    }
+}
+
+/// Core setter methods relevant to any routing type.
+impl<T: AsRef<[u8]> + AsMut<[u8]>> Header<T> {
+    /// Set the next header field.
+    #[inline]
+    pub fn set_next_header(&mut self, value: Protocol) {
+        let data = self.buffer.as_mut();
+        data[field::NXT_HDR] = value.into();
+    }
+
+    /// Set the option data length. Length of the Routing header in 8-octet units.
+    #[inline]
+    pub fn set_header_len(&mut self, value: u8) {
+        let data = self.buffer.as_mut();
+        data[field::LENGTH] = value;
+    }
+
+    /// Set the routing type.
+    #[inline]
+    pub fn set_routing_type(&mut self, value: Type) {
+        let data = self.buffer.as_mut();
+        data[field::TYPE] = value.into();
+    }
+
+    /// Set the segments left field.
+    #[inline]
+    pub fn set_segments_left(&mut self, value: u8) {
+        let data = self.buffer.as_mut();
+        data[field::SEG_LEFT] = value;
+    }
+
+    /// Initialize reserved fields to 0.
+    ///
+    /// # Panics
+    /// This function may panic if the routing type is not set.
+    #[inline]
+    pub fn clear_reserved(&mut self) {
+        let routing_type = self.routing_type();
+        let data = self.buffer.as_mut();
+
+        match routing_type {
+            Type::Type2 => {
+                data[4] = 0;
+                data[5] = 0;
+                data[6] = 0;
+                data[7] = 0;
+            }
+            Type::Rpl => {
+                // Retain the higher order 4 bits of the padding field
+                data[field::PAD] = data[field::PAD] & 0xF0;
+                data[6] = 0;
+                data[7] = 0;
+            }
+
+            _ => {}
```

Comment:
explicit panic if `routing_type` is not set?

Reply:
Should I panic on a value of `0` then? That is a valid routing type, though not one this code explicitly supports.

Reply:
Makes sense. Done.